migration/v0.18.x-v0.19.x/opensearch-migration-run.sh

```diff
@@ -0,0 +1,289 @@
+
+if [[ ! -f "${sops_config}" ]]; then
+    echo "Sops config does not exist, aborting."
+    exit 1
+elif [[ ! -f "${secrets}" ]]; then
+    echo "Secrets does not exist, aborting."
+    exit 1
+elif [[ ! -f "${common_defaults}" ]]; then
+    echo "Default common-config does not exist, aborting."
+    exit 1
+elif [[ ! -f "${common_config}" ]]; then
+    echo "Override common-config does not exist, aborting."
+    exit 1
+elif [[ ! -f "${sc_defaults}" ]]; then
+    echo "Default sc-config does not exist, aborting."
+    exit 1
+elif [[ ! -f "${sc_config}" ]]; then
+    echo "Override sc-config does not exist, aborting."
+    exit 1
+fi
+
+sc_merged=$(yq m -x -a overwrite "${common_defaults}" "${sc_defaults}" "${common_config}" "${sc_config}")
+
+ES_BUCKET=$(echo "${sc_merged}" | yq r - "objectStorage.buckets.elasticsearch")
+ES_PASSWORD=$(sops --config "${sops_config}" -d "${secrets}" --extract '["elasticsearch"]["adminPassword"]')
+ES_REPOSITORY=$(echo "${sc_merged}" | yq r - "elasticsearch.snapshotRepository")
+
+OS_BUCKET=$(echo "${sc_merged}" | yq r - "objectStorage.buckets.opensearch")
+OS_PASSWORD=$(sops --config "${sops_config}" -d "${secrets}" --extract '["opensearch"]["adminPassword"]')
+OS_REPOSITORY=$(echo "${sc_merged}" | yq r - "opensearch.snapshot.repository")
+OS_CLUSTERNAME=$(echo "${sc_merged}" | yq r - "opensearch.clusterName")
+
+if [[ "${OS_BUCKET}" == "${ES_BUCKET}" ]]; then
+    echo -n "ERROR: OpenSearch cannot use the same bucket as ODFE."
+    exit 1
+elif [[ "${OS_REPOSITORY}" == "${ES_REPOSITORY}" ]]; then
+    echo -n "ERROR: OpenSearch cannot use the same repository name as ODFE."
+    exit 1
+fi
+
+BASE_DOMAIN=$(echo "${sc_merged}" | yq r - "global.baseDomain")
+OSD_SUBDOMAIN=$(echo "${sc_merged}" | yq r - "opensearch.dashboards.subdomain")
+
+echo "--- Migration from ODFE to OpenSearch ---"
+echo
+
+echo "--- Delete Fluentd Elasticsearch to stop incoming logs."
+echo "ck8s ops helmfile wc -l app=fluentd delete"
+echo -n "- run? [y/N]: "
+read -r reply
+if [[ "${reply}" != "y" ]]; then
+    exit 1
+fi
+"${here}/../../bin/ck8s" ops helmfile wc -l app=fluentd delete
+echo "---"
+echo
+
+echo "--- Take final ODFE snapshot ---"
+echo "ck8s ops kubectl sc -n elastic-system exec opendistro-es-master-0 -c elasticsearch -- \\"
+echo "    curl -XPUT http://localhost:9200/_snapshot/${ES_REPOSITORY}/final?wait_for_completion=true \\"
+echo "        -u admin:<password>"
+echo -n "- run? [y/N]: "
+read -r reply
+if [[ "${reply}" != "y" ]]; then
+    exit 1
+fi
+"${here}/../../bin/ck8s" ops kubectl sc -n elastic-system exec opendistro-es-master-0 -c elasticsearch -- \
+    curl -XPUT "http://localhost:9200/_snapshot/${ES_REPOSITORY}/final?wait_for_completion=true" \
+        -u "admin:${ES_PASSWORD}"
+echo "---"
+echo
+
+echo "--- Delete ODFE. (Skip if you want to keep it for now, remove it now if you are going to reuse the subdomains.)"
+echo "ck8s ops helmfile sc -l group=opendistro delete"
+echo -n "- run? [y/s/N]: "
```

Comment:
It should be fine, but I would like some way of clarifying that `s` is the skip option.
So maybe change to something like this:

```suggestion
echo -n "- run? [y/s (skip)/N]: "
```

It's not super pretty, so if you have any other idea

Reply:
I've added it into the first line, so now it says "Skip using option 's' if..."